### PR TITLE
Added exports to JTS, WKT and GeoJSON

### PR DESCRIPTION
**Pull request by situx**
I added two libraries and a converter class to export OSM geometries as WKT and GeoJSON and to use them as JTS java objects. Also I added toString methods in the respective data classes. In my opinion a JSON export also of Users and Changesets could be interesting to implement. (is there a common format to do this?)

### CODE REVIEW

File: src/main/java/util/JTSConverter.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+				LinearRing ring = new LinearRing(coordSeq, geometryFactory);
+				return new Polygon(ring, null, geometryFactory);
+
+			} else {
+				return new LineString(coordSeq, geometryFactory);
+			}
+		} catch (IllegalArgumentException e) {
+			return null;
+		}
+	}
```
Comment (westnordost) [inline src/main/java/util/JTSConverter.java:88]: If I remember correctly, the direction of the line string matters with JTS while it does not matter in OSM. You would have to add logic here to reverse the order if necessary.

Comment (westnordost) [reply src/main/java/util/JTSConverter.java:88]: As said, StreetComplete which uses this library used to also use JTS until v7.1. Perhaps these old versions of some files may help you:

https://github.com/westnordost/StreetComplete/blob/v7.1/app/src/main/java/de/westnordost/streetcomplete/util/JTSConst.java

https://github.com/westnordost/StreetComplete/blob/v7.1/app/src/main/java/de/westnordost/streetcomplete/data/osm/ElementGeometry.java#L85

**Review by westnordost — COMMENTED** [review]
Okay, first the bad news: I will not merge this and there is nothing you can do to change this.

**Reason**: This library should only include those dependencies that are necessary. It is not the task of this library to do any conversions. Generally, it is not good design to have a library depend on more things than are necessary because every user of this library would then also have for example JTS, a library that is not exactly small.

I am sure that you made these changes with some (your) use case in mind, so I will comment a bit on that in an attempt to help you achieve your goal with clean code.
Basically, you can write a static converter class that takes element geometries and returns JTS geometry. Your JTSConverter class is pretty much already doing this job, but you are making one big architectural mistake:

The nodes included in OSM Ways and OSM Relations are not included with geometry but only their IDs. Querying the OSM API for every node is **massively** inefficient (each is a single HTTP request).
When you get the data from the `mapDao.getMap(boundingBox, myMapDataHandler);`, all the (node) data you need to construct the actual geometry for the included ways and relations are already included. All you need to do is to save all the nodes you get into some `Map<Long,Node>` (node id -> Node) to be able to later access it.

In other words, it is not a good idea to try to directly convert the OSM data structures `Way` and `Relation` to GeoJSON, WKT or JTS because the actual geometry is missing. Instead, use an intermediate data structure which you can directly construct out of the Node,Way,Relation data you get from the OSM API.

**Comment by westnordost** [comment]
One clue additionally. If you are just out for getting map data from the OSM API and converting it to JTS/GeoJSON, there are other libraries which do exactly that. See:
https://wiki.openstreetmap.org/wiki/Osm4j
https://wiki.openstreetmap.org/wiki/Osmosis

**Comment by situx** [comment]
Thank you for your comments and suggestions of which I can surely profit in further development. I was using your library because it was in my view one of the easiest to get access to the data I needed. If you have no interest in merging it then I will just leave my modified version in my Github and use it further.

**Pull request closed without merging**